File: app/src/main/java/com/rolandvitezhu/todocloud/datasynchronizer/BaseDataSynchronizer.java
package com.rolandvitezhu.todocloud.datasynchronizer;

import com.rolandvitezhu.todocloud.app.AppController;
import com.rolandvitezhu.todocloud.datastorage.DbLoader;

import java.util.Objects;

import javax.inject.Inject;
import javax.inject.Singleton;

import retrofit2.Retrofit;

@Singleton
public class BaseDataSynchronizer {

  @Inject
  DbLoader dbLoader;
  @Inject
  Retrofit retrofit;

  int nextRowVersion;

  @Inject
  public BaseDataSynchronizer() {
    Objects.requireNonNull(AppController.Companion.getInstance()).getAppComponent().inject(this);
  }
}

File: app/src/main/java/com/rolandvitezhu/todocloud/datasynchronizer/BaseDataSynchronizer.kt
package com.rolandvitezhu.todocloud.datasynchronizer

import com.rolandvitezhu.todocloud.app.AppController.Companion.instance
import com.rolandvitezhu.todocloud.datastorage.DbLoader
import retrofit2.Retrofit
import java.util.*
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
open class BaseDataSynchronizer @Inject constructor() {

    @Inject
    lateinit var dbLoader: DbLoader

    @Inject
    lateinit var retrofit: Retrofit

    var nextRowVersion = 0

    init {
        Objects.requireNonNull(instance)?.appComponent?.inject(this)
    }
}